src/tools/rbd/action/Trash.cc

```diff
@@ -38,7 +38,8 @@ static const std::string EXPIRED_BEFORE("expired-before");
 static const std::string THRESHOLD("threshold");
 
 static bool is_not_trash_user(const librbd::trash_image_info_t &trash_info) {
-  return trash_info.source != RBD_TRASH_IMAGE_SOURCE_USER;
+  return (trash_info.source != RBD_TRASH_IMAGE_SOURCE_USER &&
+          trash_info.source != RBD_TRASH_IMAGE_SOURCE_REMOVE);
```

Comment:
It might be confusing for users when they see an image they are removing in trash. As for me it is perfectly fine that users could only see such images running `trash list` with `--all` flag. The things will become much less confusing when we add "source" column in `trash list` output.

Note, I had already have our internal discussion trying to explain what actually this PR means, and how it differs from just moving to trash.

Reply:
I just don't want a failed image removal to get lost. The other alternative would be to always include ```RBD_TRASH_SOURCE_IMAGE_REMOVE```-tagged images in the regular list results.

Reply:
As a user I think I would prefer to have them in the regular list results, and this would be the same behavior as we had earlier. But as a programmer I am not so sure. So as you wish. I will be fine with any approach.

Reply:
Locally I modified the image list API method to include the trash images and for "rbd ls" I think it works nicely. However, once you start playing around with the image via the CLI, you start hitting "ENOENT" errors for all the actions that take images. Perhaps that's really no worse than any previous partially removed image. 

I had also modified the "trash_list" API method to exclude these "removing" images since it would be weird to double-list them between the two APIs. 

Thoughts?

Reply:
I like this. Also I think it would be nice (now or as a future improvement) to modify `rbd info` or `rbd status` to look for the image in the trash if ENOENT was returned and output something like "removing in progress or interrupted" if it is found here. It would be a nice hint for users to figure out why the image is seen in the list but can't be opened.

Reply:
update pushed